server/handlers/k8s_terminal_handler.go

```diff
@@ -0,0 +1,366 @@
+// Package handlers : collection of handlers (aka "HTTP middleware")
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/meshery/meshery/server/models"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+	CheckOrigin: func(r *http.Request) bool {
+		// TODO: Configure proper CORS in production
+		return true
+	},
+}
+
+// TerminalMessage represents WebSocket messages for terminal
+type TerminalMessage struct {
```

Comment:
Let's be schema-driven in our development by separately defining these structs as an OpenAPI schema here - https://github.com/meshery/schemas